src/config/link.rs

```diff
@@ -12,6 +12,9 @@ pub enum LinkType {
     Script,
     Cert,
     Key,
+    ReflectionWithFileName,
+    ReflectionWithService,
+    ReflectionAllFiles,
```

Comment:
do we really need 3 different type for reflection? 

Feels like ReflectionAllFiles could solve all of the problems. And if you have concerns about redundant data loaded we can load only services that are used across `@grpc` directive

Reply:
grpc directive could be defined anywhere.. it would be difficult to cover all cases specially if we make any changes in future..

This will also increase time to compile the config.. we'll have to iterate for n^2 twice (once for conversion to blueprint and other to search all grpc directives)

Reply:
could you explain why we need 3 different types for this and how the issue you've mentioned is solved?

Reply:
We are directly asking for pkg id or file name as link id.. so we just have to make request for specific files defined by the user instead of iterating all over config to find grpc method.. and thats’s the purpose of link right?

here is correction for my comment above: the complexity will be 2^n as grpc is field directive. In that case, I dont think we’ll need to change code in future but again but schema compilation will be slower. 

But at this point, I dont think iterating over fields will take A LOT more time, either of the approaches are fine. Lmk what do you think